### PR TITLE
Added Android Beginner and Intermediate Courses

### PR DESCRIPTION
**Pull request by ikayz**
The courses added are for beginners and intermediates: two Android basics courses and one intermediate Jetpack Compose course from the Google Developers Training team.

## What does this PR do?
Add resource(s)

### Description
Android Basics in Kotlin and Compose. These are well-written courses for beginners and intermediates as they teach the modern Android development approach.

### Why is this valuable (or not)?
The resources are valuable for beginners starting Android development and looking for free resources. 

### How do we know it's really free?
It's by the Google Developers Training team that offers free courses on the official Android docs website.

### For book lists, is it a book? For course lists, is it a course? etc.
They are courses

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by ikayz — COMMENTED** [review]
Fixed the sorting error

**Review by ikayz — COMMENTED** [review]
Swapped lines 211 and 210, and 213 and 212

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -208,10 +208,12 @@
 * [Android App Development for Beginners Playlist](https://www.youtube.com/playlist?list=PL6gx4Cwl9DGBsvRxJJOzG4r4k_zLKrnxl) - Bucky Roberts (thenewboston)
 * [Android App Development Tutorial for Beginners (2020) \| Complete Guides and Courses](https://www.youtube.com/playlist?list=PLwhVruPHD9rz2MwQuYSenQ0WK1IGP5rYM) - tutorialsEU
 * [Android Basics: Data Storage](https://www.udacity.com/course/android-basics-data-storage--ud845) (Udacity)
+* [Android Basics in Kotlin](https://developer.android.com/courses/android-basics-kotlin/course) - Google Developers Training team (Android)
```
Comment (eshellman) [inline courses/free-courses-en.md:211]: I think we'll move the kotlin courses to be a subcategory under android, but for now, put the kotlin course in kotlin.

Comment (eshellman) [reply courses/free-courses-en.md:211]: "This course is out of date and no longer maintained."

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -208,10 +208,12 @@
...
 * [Android Basics: Multiscreen Apps](https://www.udacity.com/course/android-basics-multiscreen-apps--ud839) (Udacity)
 * [Android Basics: Networking](https://www.udacity.com/course/android-basics-networking--ud843) (Udacity)
 * [Android Basics: User Input](https://www.udacity.com/course/android-basics-user-input--ud836) (Udacity)
 * [Android Basics: User Interface](https://www.udacity.com/course/android-basics-user-interface--ud834) (Udacity)
+* [Android Basics with Compose](https://developer.android.com/courses/android-basics-compose/course) - Google Developers Training team (Android)
```
Comment (eshellman) [inline courses/free-courses-en.md:216]:
```suggestion
* [Android Basics with Compose](https://developer.android.com/courses/android-basics-compose/course) - Google Developers Training
```

File: courses/free-courses-en.md [hunk 3]
```diff
@@ -226,6 +228,7 @@
 * [Firebase in a Weekend: Android](https://www.udacity.com/course/firebase-in-a-weekend-by-google-android--ud0352) (Udacity)
 * [Gradle for Android and Java](https://www.udacity.com/course/gradle-for-android-and-java--ud867) (Udacity)
 * [Jetpack Compose](https://www.youtube.com/playlist?list=PLQkwcJG4YTCSpJ2NLhDTHhi6XBNfk9WiC) - Philipp Lackner
+* [Jetpack Compose for Android Developers](https://developer.android.com/courses/jetpack-compose/course) - Google Developers Training team (Android)
```
Comment (eshellman) [inline courses/free-courses-en.md:231]:
```suggestion
* [Jetpack Compose for Android Developers](https://developer.android.com/courses/jetpack-compose/course) - Google Developers Training
```

File: courses/free-courses-en.md (hunk 1 above)
Comment (eshellman) [reply courses/free-courses-en.md:211]: remove it

**Review by eshellman — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 1 above)
Comment (ikayz) [reply courses/free-courses-en.md:211]: Alright, I'll remove the Kotlin course. Thanks for pointing that out. Only the Jetpack Compose courses will remain

**Review by ikayz — COMMENTED** [review]
The suggested change of removing the Android Basics in Kotlin has been applied

**Comment by ikayz** [comment]
@eshellman the requested changes have been applied

**Pull request merged**